decidim/decidim
# Fix attachments when called from Cells

#### :tophat: What? Why?

While working on Votings, it was detected an edge case on Cells and Attachments. 

Basically the problem is that after Rails 6.1 upgrade (see #8610) we dropped the extension from the partial calls at https://github.com/decidim/decidim/blob/a15c91cd006fec147d3c4705173e3ab0b9296923/decidim-core/app/views/decidim/application/_attachments.html.erb#L4.

Most of the times it isn't a problem, but in votings we have the `AttachmentsAndFoldersCell` calling attachments from a cell, and as cells don't have the HTML extension in their file it doesn't find the partial giving an exception: 


<details>
<summary>Click for traceback</summary>

  
ActionView::Template::Error (Template missing: view: `_attachments.erb` prefixes: ["/home/apereira/Work/decidim/decidim/development_app/app/cells/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-assemblies/app/cells/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-assemblies/app/views/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-conferences/app/cells/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-conferences/app/views/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-core/app/cells/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-core/app/cells/amendable/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-core/app/views/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-admin/app/cells/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-admin/app/views/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-forms/app/cells/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-verifications/app/cells/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-participatory_processes/app/cells/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-participatory_processes/app/views/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-comments/app/cells/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-comments/app/views/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-proposals/app/cells/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-proposals/app/views/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-meetings/app/cells/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-meetings/app/views/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-debates/app/cells/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-debates/app/views/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-budgets/app/cells/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-budgets/app/views/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-accountability/app/cells/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-accountability/app/views/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-sortitions/app/cells/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-sortitions/app/views/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-blogs/app/cells/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-blogs/app/views/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-consultations/app/cells/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-consultations/app/views/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-elections/app/cells/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-elections/app/views/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-elections/app/cells/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-elections/app/views/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-initiatives/app/cells/decidim/application", "/home/apereira/Work/decidim/decidim/decidim-initiatives/app/views/decidim/application"]):
    10: %>
    11: 
    12: <% landing_content_blocks.each do |content_block| %>
    13:   <%= cell content_block.manifest.cell, content_block %>
    14: <% end %>
    15: 
    16: <% if landing_content_blocks.empty? %>
  
cells (4.1.7) lib/cell/view_model.rb:167:in `find_template'
cells (4.1.7) lib/cell/view_model.rb:109:in `render_to_string'
cells (4.1.7) lib/cell/layout.rb:23:in `render_to_string'
cells (4.1.7) lib/cell/view_model.rb:104:in `render'
/home/apereira/Work/decidim/decidim/decidim-core/app/helpers/decidim/attachments_helper.rb:13:in `attachments_for'
/home/apereira/Work/decidim/decidim/decidim-elections/app/cells/decidim/votings/content_blocks/landing_page/attachments_and_folders_cell.rb:16:in `show'
cells (4.1.7) lib/cell/view_model.rb:114:in `render_state'
cells (4.1.7) lib/cell/caching.rb:46:in `render_state'
cells (4.1.7) lib/cell/view_model.rb:92:in `call'
cells-rails (0.1.4) lib/cell/rails.rb:55:in `call'
/home/apereira/Work/decidim/decidim/decidim-core/lib/decidim/view_model.rb:33:in `block in call'
/home/apereira/Work/decidim/decidim/decidim-core/lib/decidim/view_model.rb:51:in `block in instrument'
activesupport (6.1.4.7) lib/active_support/notifications.rb:203:in `block in instrument'
activesupport (6.1.4.7) lib/active_support/notifications/instrumenter.rb:24:in `instrument'
activesupport (6.1.4.7) lib/active_support/notifications.rb:203:in `instrument'
/home/apereira/Work/decidim/decidim/decidim-core/lib/decidim/view_model.rb:50:in `instrument'
/home/apereira/Work/decidim/decidim/decidim-core/lib/decidim/view_model.rb:32:in `call'
cells (4.1.7) lib/cell/view_model.rb:125:in `to_s'
actionview (6.1.4.7) lib/action_view/buffers.rb:29:in `<<'
/home/apereira/Work/decidim/decidim/decidim-elections/app/views/decidim/votings/votings/show.html.erb:13
/home/apereira/Work/decidim/decidim/decidim-elections/app/views/decidim/votings/votings/show.html.erb:12:in `each'
/home/apereira/Work/decidim/decidim/decidim-elections/app/views/decidim/votings/votings/show.html.erb:12


</details>


The fix is just some renaming in these files. I've also added a regressions spec for this particular cell. 

#### :pushpin: Related Issues

- Related to https://github.com/decidim/decidim/pull/9112#issuecomment-1087412044

#### Testing

1. Run the spec before and after this PR

or

1. Seed the database
2. Go to a voting
3. See the exception 

:hearts: Thank you!
